packages/xo-web/src/common/xo/index.js

```diff
@@ -527,29 +526,39 @@ export const addHostToPool = (pool, host) => {
       })
     )
   }
+}
 
-  return confirm({
+import AddHostsModalBody from './add-hosts-modal' // eslint-disable-line import/first
+export const addHostsToPool = pool =>
+  confirm({
     icon: 'add',
-    title: _('addHostModalTitle'),
-    body: <AddHostModalBody pool={pool} />,
+    title: _('addHostsModalTitle'),
+    body: <AddHostsModalBody pool={pool} />,
   }).then(params => {
-    if (!params.host) {
-      error(_('addHostNoHost'), _('addHostNoHostMessage'))
+    const { hosts } = params
+    if (isEmpty(hosts)) {
+      error(_('addHostsNoHosts'), _('addHostsNoHostsMessage'))
       return
     }
-    return _call('pool.mergeInto', {
-      source: params.host.$pool,
-      target: pool.id,
-      force: true,
-    }).catch(error => {
+
+    return Promise.all(
+      map(hosts, host =>
```

Comment:
This should be done on the server side and probably sequentially to avoid any issue or we should at least thoroughly test it if we want to do it simultaneously.